classes/Cart.php

```diff
@@ -2404,7 +2404,7 @@ public function getProductAddressId($product)
         }
 
         // Get delivery address of the product from the cart
-        if (!$this->addressFactory->addressExists($addressId)) {
+        if (!$this->addressFactory->addressExists($addressId, true)) {
```

Comment:
Make sure that the cache is cleaned when address is deleted.

FYI : https://github.com/PrestaShop/PrestaShop/pull/21616